tests/test.sh

```diff
@@ -0,0 +1,172 @@
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+
+    if [ "$ENV" = "minikube" ]; then
+        # ToDo: Clean up, I'm not happy about having specific checks for the stacks with multiple ports
+        # But since we're testing against minikube, we need to specifically create the URL/ingress before pushing
+        # And if there's multiple ports in the devfile, a port must be specified.
+        if [ "$devfileName" = "java-wildfly" ] || [ "$devfileName" = "java-wildfly-bootable-jar" ]; then
+            $ODO_PATH url create --host "$(minikube ip).nip.io" --port 8080 || error=true
+            $ODO_PATH url create --host "$(minikube ip).nip.io" --port 16686 || error=true
+        else
+            $ODO_PATH url create --host "$(minikube ip).nip.io" || error=true
+        fi
+        if $error; then
+            echo "ERROR url create failed"
+            $ODO_PATH project delete -f "$devfileName"
+            FAILED_TESTS="$FAILED_TESTS $devfileName"
+            return 1
+        fi
+    fi
+
+    $ODO_PATH push || error=true
+    if $error; then
+        echo "ERROR push failed"
+        $ODO_PATH delete -f -a || error=true
+        $ODO_PATH project delete -f "$devfileName"
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+
+    # check if application is responding
+    urls=$(getURLs)
+
+    for url in $urls; do
+        statusCode=200
+        
+        waitForHTTPStatus "$url" "$statusCode"
+        if [ $? -ne 0 ]; then
+            echo "ERROR unable to get working url"
+            $ODO_PATH project delete -f "$devfileName"
+            FAILED_TESTS="$FAILED_TESTS $devfileName"
+            error=true
+            return 1
+        fi
+    done
+
+    # kill -9 $CPID
+    $ODO_PATH delete -f -a || error=true
+    $ODO_PATH project delete -f "$devfileName"
+
+    if $error; then
+        echo "FAIL"
+        # record failed test
+        FAILED_TESTS="$FAILED_TESTS $devfileName"
+        return 1
+    fi
+
+    return 0
+}
+
+if [ -z $ODO_PATH ]; then
+  ODO_PATH=odo
+fi
+if [ -z $YQ_PATH ]; then
+  YQ_PATH=yq
+fi
+
+if [ -z $REGISTRY ]; then
+  REGISTRY=local
+fi
+if [ "$REGISTRY" != "local" ] && [ "$REGISTRY" != "remote" ]; then
+  echo "ERROR: Allowed values for REGISTRY are either \"local\" (default) or \"remote\"."
+  exit 1
+fi
+
+
+# If $STACK is set, test only that specific stack
+# Otherwise, test the other stacks.
+if [ $STACK ]; then
+  stackFolder="$(realpath $(dirname $STACK))"
+  devfile_name="$(basename $STACK)"
+  devfile_path=$stackFolder/$devfile_name/devfile.yaml
+  test "$devfile_name" "$devfile_path"
+else
+  for devfile_dir in $(find $DEVFILES_DIR -maxdepth 1 -type d ! -path $DEVFILES_DIR); do
+      devfile_name="$(basename $devfile_dir)"
+      devfile_path=$devfile_dir/devfile.yaml
+
+      test "$devfile_name" "$devfile_path"
+  done
+fi
+
+
+# remember if there was an error so the script can exist with proper exit code at the end
+error=false
```

Comment:
Do we need this flag?  It doesn't seem to be used to test conditions in subsequent commands.  I think there's a typo in the comment -  `exist` should be `exit` but irrelevant if this line is not needed.

Reply:
Isn't it used else in the `test` function specifically (e.g. https://github.com/redhat-developer/devfile-registry/pull/11/files#diff-dad7b84674fccc140b0fb2f17af742872a2f5cda5ddd36af501f39b404baf0cbR90)

Reply:
Actually, I see what you mean. Its use is scoped to only the `test` function, and isn't needed outside of it.

We can keep the variable as-is in the `test` function, and remove the references elsewhere, as it's not needed.

Reply:
Updated.